src/nations/nations.cpp

```diff
@@ -1245,6 +1245,19 @@ void cleanup_nation(sys::state& state, dcon::nation_id n) {
 		state.world.delete_wargoal(wg);
 	}
 
+	// If the nation stops existing we shall cleanup all wars it was involved at
+	// War participation may already be annulled by previous calls so we have to iterate
```

Comment:
This shouldn't happen. If the nation or the war hasn't been deleted, then everything should remain in place. Instead what you want to do is something like this:
```
par = ... the range or war participation ...
while(par.begin() != par.end()) {
	... do something that will result in the deletion of (*par.begin()) ...
}
```
However, there is a big problem here
Look at how a peace offer is implemented. We go over each wargoal in the offer one by one, implementing them. Suppose that annexing the war leader happens to be first on the list. Your code here would then, in implementing that wargoal, delete the war itself, thereby deleting all the wargoals attached to it, thereby preventing any other wargoals from being implemented, even if they were directed against other people. It also has the problem that the peace offer routine is going to continue and read out information from this now deleted war to possibly try to end it on its own. We could end up trying to delete the war twice, essentially